crates/freeze/src/types/partitions.rs

```diff
@@ -388,6 +404,7 @@ impl Partition {
     }
 
     /// return Vec of dimensions defined in partitions
+    // NOTE: this function is not exhaustive
```

Comment:
what is meant by these notes?

Reply:
The commented points are branches that take or return `Dim` but does not emit type error even all of its variants are properly handled. For example, adding a variant for `Dim` does not guarantee `Dim::all_dims()` function to return every variant of the enum because the programmer would have to manually write the additional variant. These codes - for me - felt error-prone because a new commiter is hard to know every part of the codebase that requires additional code for added variant. I meant this for `not exhaustive`.